### DISCUSSION TITLE
Unstable user count measurement in client.shard.broadcastEval()

### DISCUSSION THREAD

**Discussion by Taka005** [discordjs/discord.js#10233] [Q&A]
### Which package is this bug report for?

discord.js

### Issue description

I am using client.shard.broadcastEval() to get user and server totals for all shards. (from the code below), but the client.shard.broadcastEval() value is unstable at times and is not returning accurate results.
From my research, this bug is more likely to occur the longer it has been since the BOT was launched.
This bug prevents accurate measurement of statistics
My BOT has 3 shards up and running. This was occurring even with two shards.

### Code sample

```typescript
(await client.shard.broadcastEval(c=>c.guilds.cache.size))
      .reduce((a,c)=>a+c,0);
// returning value is [null,1200,1001]
// For some reason, a null value comes in.

(await client.shard.broadcastEval(c=>c.guilds.cache.reduce((a,g)=>a+g.memberCount,0)))
      .reduce((a,c)=>a+c,0);
// Same here.
```

### Versions

discord.js 14.14.1
Node.js 18.16.0
Ubuntu Server 22.04.4

### Issue priority

High (immediate attention needed)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildMembers, GuildPresences, GuildMessages, MessageContent

**Comment by Qjuh** [comment]
This sounds like you might have an unavailable Guild on your first shard. So there is a Guild in your `<Client>.guilds.cache` with `available: false` property causing its `memberCount` property to be undefined. Your reduce call then causes the whole sum to become `NaN` which gets JSONified into `null` and sent back over the IPC channel to where you call the broadcastEval.
So not a bug in discord.js itself but in your handling of unavailable Guilds. filter the guilds.cache before the reduce calls and you should be good.

**Comment by Taka005** [comment]
> This sounds like you might have an unavailable Guild on your first shard. So there is a Guild in your `<Client>.guilds.cache` with `available: false` property causing its `memberCount` property to be undefined. Your reduce call then causes the whole sum to become `NaN` which gets JSONified into `null` and sent back over the IPC channel to where you call the broadcastEval. So not a bug in discord.js itself but in your handling of unavailable Guilds. filter the guilds.cache before the reduce calls and you should be good.

I understand the fix about counting users. Also, all guild counts are a little higher than the actual number of guilds when there are unavailable guilds.
```js
(await client.shard.broadcastEval(c=>c.guilds.cache.size))
      .reduce((a,c)=>a+c,0);
      
 // The number of all guilds is 3100, but when some guilds are not available, the value is 3109, etc.
```

**Reply by Qjuh** [reply]
Because you didn’t filter out unavailable Guilds. And/or you maybe fetched Guilds not belonging to that shard.

 You should *never* fetch guilds or channels, since they all are cached anyway (exception: archived threads or DMChannels (but the latter are handled on shard 0, so shouldn’t be fetched anywhere else either)).
With Guilds intent all guilds and channels belonging to that shard are cached.
